### PR TITLE
Contributor guide: add info about gitlab repository for data

### LINKED ISSUE #4889: Add a note in our documentation/on website saying that we have a gitlab repo
## Description

It seems that we do not have any reference to our gitlab repository. This page https://gitlab.com/scikit-image must appear in our documentation, I guess the one for dev and contributors. We must say that images must be submitted to this repository.

### PR DESCRIPTION
**Pull request by sciunto**
## Description

closes #4889 

Result here: https://139-2014929-gh.circle-artifacts.com/0/doc/build/html/contribute.html#id4

### CODE REVIEW

File: CONTRIBUTING.txt [hunk 1]
```diff
@@ -46,6 +46,15 @@ Here's the long and short of it:
      - ``upstream``, which refers to the ``scikit-image`` repository
      - ``origin``, which refers to your personal fork
 
+.. note::
+
+    Although our code is hosted on `github
+    <https://github.com/scikit-image/>`_, our database is stored on `gitlab
```
Comment (emmanuelle) [inline CONTRIBUTING.txt:52]:
```suggestion
    <https://github.com/scikit-image/>`_, our datasets are stored on `gitlab
```

Comment (emmanuelle) [reply CONTRIBUTING.txt:52]: database sounds like we have a big database like mysql etc.

**Comment by sciunto** [comment]
(rebased to observe circle-ci)

File: CONTRIBUTING.txt (hunk 1 above)
Comment (jni) [reply CONTRIBUTING.txt:52]: @sciunto you've marked this as resolved but I still see database?

File: CONTRIBUTING.txt [hunk 2]
```diff
@@ -46,6 +46,15 @@ Here's the long and short of it:
...
+    <https://gitlab.com/scikit-image>`_. and fetched with `pooch
```
Comment (jni) [inline CONTRIBUTING.txt:53]: Add full link? I think it's

```suggestion
    <https://gitlab.com/scikit-image/data>`_ and fetched with `pooch
```

File: CONTRIBUTING.txt (hunk 1 above)
Comment (sciunto) [inline CONTRIBUTING.txt:52]:
```suggestion
    <https://github.com/scikit-image/>`_, our dataset is stored on `gitlab
```

**Comment by sciunto** [comment]
> @sciunto I've reiterated one suggestion from Emma 

I committed the suggestion from github and then forgot to pull before rebasing the branch. :facepalm:

**Comment by emmanuelle** [comment]
Thanks @sciunto, merging!

**Pull request merged**